### PR TITLE
Hilbert Curve FreeCAD Macro

### PR DESCRIPTION
**Pull request by pysimone**
Thank you for creating a pull request to contribute to FreeCAD-macros!
To integrate your macro please make sure the following steps are complete:

- [ ] Please check this box if you're not submitting a new macro.   
- [x] Are you submitting a new macro ?  
- [x] Have you followed the ['How to submit a macro'](../README.md#how-to-submit-a-macro) section of the README.md ?  
- [x] Your macro has a [Description](../README.md#macro-description) in its header.  
- [x] Your macro has a [CamelCase name](../README.md#camelcase-macro-name).  
- [x] Your macro is named [appropriately](../README.md#macro-name-specifics).  
- [x] Your macro contains a [Metadata section](../README.md#macro-metadata) that immediately follows the header description.  
- [x] Your macro is Python3/Qt5 compliant and tested on the latest FreeCAD stable and development releases.  
- [ ] You're including documentation on how your macro works (bonus: screenshots and/or video on the Wiki)  
- [x] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)  
- [ ] Commit message is titled in the following way `[MacroName] Short description`.
- [ ] Optional, write or update the changelog in the macro, from latest to oldest.

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

### CODE REVIEW

File: HilbertCurve.FCMacro [hunk 1]
```diff
@@ -0,0 +1,246 @@
+# -*- coding: utf-8 -*-
```
Comment (galou) [inline HilbertCurve.FCMacro:1]: Can you please move to `ObjectCreation/HilbertCurve.FCMacro`?

File: HilbertCurve.FCMacro [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+# USA
+#
+# You can contact me by mail at simone.bignetti@linux.it
+
+__Name__ = 'HilbertCurve'
+__Comment__ = 'This macro creates a Hilbert Curve wire in 2 or 3 dimensions with many iterations.'
+__Author__  = 'Simone Bignetti'
+__Version__ = '01.01'
```
Comment (galou) [inline HilbertCurve.FCMacro:27]: __Version__ = '1.1.0'

File: HilbertCurve.FCMacro [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+the number of the iterations and the length of the wire segment.
+Click OK to create the wire or CANCEL to exit the macro.
+You can use the just created wire to do a sweep in the part Workbench, But first of all
+it's better to apply a radius to the wire, or the sweep will be badly formed.
+The size of the radius depends on the length chosen for the curve segments and
+on the shape of the profile you want to sweep.
+Maybe you need to take a few tries."""
+__Status__ = 'Stable'
+__Requires__ = ''
+__Communication__ = ''
```
Comment (galou) [inline HilbertCurve.FCMacro:45]: Would you have a forum link to add here?

File: HilbertCurve.FCMacro [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+__Files__ = ''
```
Comment (galou) [inline HilbertCurve.FCMacro:46]: An icon would be nice.

File: HilbertCurve.FCMacro [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+
+import FreeCAD, Draft  # For the wire
+
+from PySide import QtGui, QtCore  # For the gui
+
+class HilbertCurve:  # The class of the Hilbert's Curve.
```
Comment (galou) [inline HilbertCurve.FCMacro:53]: Docstring should be on the following line.

File: HilbertCurve.FCMacro [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+The size of the radius depends on the length chosen for the curve segments and
+on the shape of the profile you want to sweep.
+Maybe you need to take a few tries."""
+__Status__ = 'Stable'
+__Requires__ = ''
+__Communication__ = ''
+__Files__ = ''
+
+
+import FreeCAD, Draft  # For the wire
```
Comment (galou) [inline HilbertCurve.FCMacro:49]: - Imports on separate lines
- `import FreeCAD as app` for consistency with other macros.

**Review by galou — CHANGES_REQUESTED** [review]
Thanks for contributing!

File: HilbertCurve.FCMacro (hunk 1 above)
Comment (pysimone) [reply HilbertCurve.FCMacro]: moved

File: HilbertCurve.FCMacro (hunk 2 above)
Comment (pysimone) [reply HilbertCurve.FCMacro]: updated

File: HilbertCurve.FCMacro (hunk 3 above)
Comment (pysimone) [reply HilbertCurve.FCMacro]: Added the link to the forum/macors, where I told about this macro.

File: HilbertCurve.FCMacro (hunk 4 above)
Comment (pysimone) [reply HilbertCurve.FCMacro]: Uploaded the icon and updated __icon__

File: HilbertCurve.FCMacro (hunk 5 above)
Comment (pysimone) [reply HilbertCurve.FCMacro]: Done many changes to the docstrings

File: HilbertCurve.FCMacro (hunk 6 above)
Comment (pysimone) [reply HilbertCurve.FCMacro]: done

**Review by pysimone — COMMENTED** [review]
I did the required changes

**Comment by galou** [comment]
Thanks!

**Comment by pysimone** [comment]
Thank you very much for your attention!

**Pull request merged**